umap/static/umap/js/modules/utils.js

```diff
@@ -446,3 +448,176 @@ export function eachElement(selector, callback) {
     callback(el)
   }
 }
+
+export class WithEvents {
```

Comment:
Why not reusing the standard `addEventListener`/ `CustomEvent`+`dispatchEvent` while maintaining the `on`/`fire` API if you prefer?